### PR TITLE
Drop support for old visual studio versions

### PR DESCRIPTION
**Pull request by pepone**
This PR updates the Windows build to require Visual Studio 2022, older versions are no longer supported. The PHP mapping is no longer built on Windows.

### CODE REVIEW

File: cpp/BUILDING.md [hunk 1]
```diff
@@ -303,11 +301,6 @@ Open the Visual Studio solution that corresponds to the Visual Studio version
 you are using.
 
 - For Visual Studio 2022 use [msbuild/ice.v143.sln](./msbuild/ice.v143.sln)
```
Comment (externl) [inline cpp/BUILDING.md:303]: Can we rename this file to `ice.sln`? Is it really standard/required to have one solution per visual studio version?

File: php/BUILDING.md [hunk 2]
```diff
@@ -3,16 +3,14 @@
...
 
 ZeroC provides [binary distributions] for many platforms, including
-Linux and Windows, so building Ice for PHP from source is usually
-unnecessary.
+Linux, so building Ice for PHP from source is usually unnecessary.
 
 * [PHP Build Requirements](#php-build-requirements)
 * [Building the PHP Extension](#building-the-php-extension)
   * [Linux or macOS](#linux-or-macos)
   * [Windows](#windows)
```
Comment (externl) [inline php/BUILDING.md:11]: This should be removed from the toc

**Review by externl — APPROVED** [review]
Looks good!

File: php/BUILDING.md (hunk 2 above)
Comment (externl) [reply php/BUILDING.md:11]: GitHub highlighted the wrong line, I was trying to show `* [Windows](#windows)` in the table of contents.

File: cpp/BUILDING.md (hunk 1 above)
Comment (pepone) [reply cpp/BUILDING.md:303]: We can, this was required before because we didn't build the same projects with all versions.

File: cpp/BUILDING.md [hunk 3]
```diff
@@ -37,9 +37,7 @@ unnecessary.
 Ice was extensively tested using the operating systems and compiler versions
 listed on [supported platforms][2].
 
-On Windows, the build requires a recent version of Visual Studio. When building
-with Visual Studio 2017 you have to install Desktop development with C++ workload
-and select the optional component Windows 8.1 SDK and UCRT SDK.
+On Windows, the build requires a recent version of Visual Studio 2022.
```
Comment (bernardnormier) [inline cpp/BUILDING.md:40]: Should be either "requires Visual Studio 2022" or "requires Visual Studio 2022 version XY or greater".

**Review by bernardnormier — APPROVED** [review]

**Pull request merged**